sirepo/sim_data/ml.py

```diff
@@ -62,3 +63,57 @@ def _lib_file_basenames(cls, data):
         if name:
             return [cls.lib_file_name_with_model_field("dataFile", "file", name)]
         return []
+
+
+class ArchiveManager:
+    import contextlib
+
+    _ARCHIVE_EXTENSIONS = (
+        ".tar.gz",
+        ".zip",
+    )
+
+    def __init__(self, file_path):
+        import zipfile
+        import tarfile
+
+        self.file_path = file_path
+        self.filename = file_path if isinstance(file_path, str) else file_path.basename
+        if self._is_archive_type(".zip"):
```

Comment:
self.pkupdate would work if ArchiveManager was a subclass of PKDict.  Why not use _ARCHIVE_EXTENSIONS possibly in a loop checking the archives that are supported?

Reply:
[6e561ce](https://github.com/radiasoft/sirepo/pull/4736/commits/6e561ced430aef5c3176897462da7003d538df7e)

I think this better...???